wallet/wallet.c

```diff
@@ -2235,16 +2235,18 @@ bool wallet_network_check(struct wallet *w,
 		db_stmt_done(stmt);
 		if (!bitcoin_blkid_eq(&chainhash,
 				      &chainparams->genesis_blockhash)) {
+			const tal_t *ctx = tal(NULL, char);
 			log_broken(w->log, "Wallet blockchain hash does not "
 					   "match network blockchain hash: %s "
 					   "!= %s. "
 					   "Are you on the right network? "
-					   "(--network={bitcoin, testnet, regtest, "
-					   "litecoin or litecoin-testnet})",
+					   "(--network={one of %s})",
 				   type_to_string(w, struct bitcoin_blkid,
 						  &chainhash),
 				   type_to_string(w, struct bitcoin_blkid,
-						  &chainparams->genesis_blockhash));
+						  &chainparams->genesis_blockhash),
+				   chainparams_get_network_names(ctx));
```

Comment:
Use `tmpctx` instead here, it should be in scope if you `#include<common/utils.h>` (which is probably already included in this file somewhere, basically just check if `tmpctx` is used anywhere else).

This is a temporary context that is automatically freed in the mainloop, rather than you having to manually create and free one every time you need it.